### PR TITLE
Add ws monitor copy mode

### PR DESCRIPTION
**Pull request by TheNeikos**
This also adds various checks for safety

### CODE REVIEW

**Comment by neithernut** [comment]
ACK.

File: src/compositor/monitor.c [hunk 1]
```diff
@@ -211,12 +218,30 @@ ws_monitor_set_mode_with_id(
...
+
 struct ws_monitor_mode*
 ws_monitor_add_mode(
     struct ws_monitor* self,
     int width,
     int height
 ) {
+    //!< We exit rather than letting the program finish as it will hang the
+    // graphic card
+    ws_log(&log_ctx, LOG_CRIT, "Looks like we're in a doozy! This is undefined"
```
Comment (matthiasbeyer) [inline src/compositor/monitor.c:242]: You can do a `LOG_EMERG` here.

File: src/compositor/monitor.c [hunk 2]
```diff
@@ -225,6 +250,7 @@ ws_monitor_add_mode(
     mode->id = self->mode_count++;
     mode->mode.hdisplay = width;
     mode->mode.vdisplay = height;
+    mode->mode.clock = 155;
```
Comment (matthiasbeyer) [inline src/compositor/monitor.c:253]: Magic constant? Please add documentation.

File: src/compositor/monitor.h [hunk 3]
```diff
@@ -131,6 +131,9 @@ ws_monitor_set_mode_with_id(
  *
  * @memberof ws_monitor
  *
+ * @warning DO NOT USE! This is not yet safe as critical features are to be
+ * implemented and it will stop waysome to save your Graphics Card
```
Comment (matthiasbeyer) [inline src/compositor/monitor.h:135]: Wording.

File: src/compositor/monitor.c (hunk 2 above)
Comment (TheNeikos) [inline src/compositor/monitor.c:253]: No, this is still all **todo** code, so disregard that.

File: src/compositor/monitor.h (hunk 3 above)
Comment (TheNeikos) [inline src/compositor/monitor.h:135]: What's wrong with it?

Comment (matthiasbeyer) [inline src/compositor/monitor.h:135]: `"save"`?

Comment (TheNeikos) [inline src/compositor/monitor.h:135]: Yes

```
keep safe or rescue (someone or something) from harm or danger.
```

**Comment by neithernut** [comment]
As already noted: ACK.

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**